File: 406.queue-reconstruction-by-height.py
#
# @lc app=leetcode id=406 lang=python3
#
# [406] Queue Reconstruction by Height
#

# @lc code=start
class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        people_copy = people.copy()
        people_copy.sort(key=lambda x: x[0], reverse=True)
        print(people_copy)

        return people_copy
# @lc code=end

File: greedy/406.queue-reconstruction-by-height.swift
/*
 * @lc app=leetcode id=406 lang=swift
 *
 * [406] Queue Reconstruction by Height
 */

// @lc code=start
class Solution {
    func reconstructQueue(_ people: [[Int]]) -> [[Int]] {
        let sortedPeople = people.sorted {
            $0[0] == $1[0] ? $0[1] < $1[1] : $0[0] > $1[0]
        }

        var correctQueue = [[Int]]()

        for person in sortedPeople {
            correctQueue.insert(person, at: person[1])
        }
        return correctQueue
    }
}
// @lc code=end

